### PR TITLE
Fix 3510 - enable optimization.sideEffects, innerGraph and usedExports by default

### LINKED ISSUE #3510: Document webpack change: enable optimization.sideEffects, innerGraph and usedExports by default
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/10317

---

**What kind of change does this PR introduce?**
change

**Did you add tests for your changes?**
existing tests

**Does this PR introduce a breaking change?**
yes

**What needs to be documented once your changes are merged?**
optimization.sideEffects, innerGraph and usedExports is enabled by default in all modes (was disabled in development before)

### PR DESCRIPTION
**Pull request by jeffin143**
Fix #3510 - enable optimization.sideEffects, innerGraph and usedExports by default

- [x] Read and sign the [CLA][1]. PRs that haven't signed it won't be accepted.
- [x] Make sure your PR complies with the [writer's guide][2].
- [x] Review the diff carefully as sometimes this can reveal issues.
- [x] Do not abandon your Pull Request: [Stale Pull Requests][3].

### CODE REVIEW

File: src/content/configuration/optimization.md [hunk 1]
```diff
@@ -419,7 +419,7 @@ module.exports = {
 
 Tells webpack to determine used exports for each module. This depends on [`optimization.providedExports`](#optimizationoccurrenceorder). Information collected by `optimization.usedExports` is used by other optimizations or code generation i.e. exports are not generated for unused exports, export names are mangled to single char identifiers when all usages are compatible.
 Dead code elimination in minimizers will benefit from this and can remove unused exports.
-By default `optimization.usedExports` is enabled in `production` [mode](/configuration/mode/) and disabled elsewise.
+By default `optimization.usedExports` is enabled in all [mode](/configuration/mode/).
```
Comment (montogeek) [inline src/content/configuration/optimization.md]:
```suggestion
By default `optimization.usedExports` is enabled in all [modes](/configuration/mode/).
```

Comment (montogeek) [reply src/content/configuration/optimization.md]: Please change in all occurrences.

Comment (jeffin143) [reply src/content/configuration/optimization.md]: Resolved it everywhere

**Comment by montogeek** [comment]
Thanks!

**Pull request merged**